### ISSUE TITLE
Bug: File/directory pattern is reseted after I reload/reopen Atom editor

### ISSUE THREAD

**Issue by shuriktm** [atom/find-and-replace#393]
When I reload or reopen Atom editor "File/directory pattern" field is always empty.
It should contain the last used pattern for easy search in the project.

It also doesn't have a history, so I can't choose previous used patterns.

**Comment by jussiry** [comment]
Agreed. I often work with many projects at the same time, with each having different directory pattern to filter the search. Remembering these patterns over restarts would save a lot of time.

**Comment by benogle** [comment]
I opened a PR for this a while ago: https://github.com/atom/find-and-replace/pull/480. It was unclear if it would be a net positive, though.

**Comment by kumarharsh** [comment]
It's a very important feature, as I have to type the whole paths and exclusions every time I open Atom.

For example, in my sublime, this is my search path:

`D:/Workspace/code/secret-project-long-path/src/app, *.js, *.html, *.jade, -node_modules/, -.tmp/, -build/, -src/public/`

Imagine typing all this every single day. :cry: 

Also, the search paths do not treat `/` as a directory separator in windows, but sublime does it, so that's another thing you might consider supporting, but I guess I'll raise that as a separate issue.

**Comment by benogle** [comment]
@kumarharsh After reloading you can press `up` in any of the find and replace boxes to get the last value entered, even between restarts.

**Comment by kumarharsh** [comment]
Oh, thanks, I didn't know that.

Would it be possible to introduce an option to persist the last-used path on restarts?

**Comment by benogle** [comment]
That PR mentioned above does just that, but it still needs work.

**Comment by kumarharsh** [comment]
i thought that was to *keep the panel itself open across restarts*. I meant that *only the settings* should be persisted.

**Comment by benogle** [comment]
That PR will start the find panels in exactly the same state as when atom was quit. So if you have project find open with some text in there, that state will be preserved.

**Comment by jussiry** [comment]
Sometimes Atom remember directory patterns (when you press `up`) and sometimes it doesn't (this is on Linux). I find that to be way more important feature, than simply opening search panel, which you can do with single key command. I suppose for this issue (to help devs) more rigorous testing on when does it forget the patterns would be needed.

**Comment by benogle** [comment]
The find and replace state (all editor state, really) is attached to a given 'project', an opened directory. So if you open `atom ~/dir1` and search for `omgwow`, then kill it and open `atom ~/dir2` and press up in the find panel, there will be nothing. However, if you opened `atom ~/dir1` again, and pressed up in the find panel, `omgwow` would be there.

**Comment by nemoDreamer** [comment]
@benogle There seems to be an issue w/ the history, though...

For example, my last entry is _always_ `public/javascripts/, src/stylesheets, test/javascripts, !public/javascripts/jquery/`, and I keep having to correct it to `public/javascripts/, src/stylesheets, test/javascript, !test/javascript/output, !public/javascripts/jquery/` to fix that `test` typo and ignore that massive `output`...

I've tried various ways of quitting Atom, but the change never _sticks_ in memory. Is there a physical file these are stored in, that I can go tinker with?

(i guess I could hide the `.gitignore`'d `output`, but that's beside the general point)

**Comment by r-owen** [comment]
I'm also keen to see Atom remember the pattern. Otherwise it searches far too many files.
